services/cases-api/lambdas/getCaseList.js

```diff
@@ -12,42 +13,78 @@ export async function main(event) {
-    },
-    KeyConditionExpression: 'PK = :pk AND begins_with(SK, :sk)',
-  };
-
-  const [casesDbError, casesDbResponse] = await to(sendCasesRequest(params));
-  if (casesDbError) {
-    return response.failure(casesDbError);
+  const [getAllUserCasesError, userCases] = await to(getUserCases(personalNumber));
+  if (getAllUserCasesError) {
+    console.error('getAllUserCasesError', getAllUserCasesError);
+    return response.failure(getAllUserCasesError);
   }
 
-  if (!casesDbResponse.Count) {
-    return response.failure(new ResourceNotFoundError('No cases found'));
+  if (userCases.length === 0) {
+    return response.failure(new ResourceNotFoundError('No user cases found'));
   }
 
-  const cases = casesDbResponse.Items.map(item => objectWithoutProperties(item, ['PK', 'SK']));
+  const userCasesWithoutPK_SK_GSI1 = userCases.map(item =>
+    objectWithoutProperties(item, ['PK', 'SK', 'GSI1'])
+  );
 
   return response.success(200, {
     type: 'getCases',
     attributes: {
-      cases,
+      cases: userCasesWithoutPK_SK_GSI1,
     },
   });
 }
 
-async function sendCasesRequest(params) {
-  const [dynamoDbCallError, dynamoDbQueryResult] = await to(dynamoDb.call('query', params));
-  if (dynamoDbCallError) {
-    throwError(dynamoDbCallError.statusCode, dynamoDbCallError.message);
+async function getUserCases(personalNumber) {
+  const [getApplicantCasesError, applicantCasesResult] = await to(
+    getApplicantCases(personalNumber)
+  );
+  if (getApplicantCasesError) {
+    console.error('getApplicantCasesError', getApplicantCasesError);
+    throwError(getApplicantCasesError.statusCode, getApplicantCasesError.message);
   }
 
-  return dynamoDbQueryResult;
+  const [getCoApplicantCasesError, coApplicantCasesResult] = await to(
+    getCoApplicantCases(personalNumber)
+  );
+  if (getCoApplicantCasesError) {
+    console.error('getCoApplicantCasesError', getCoApplicantCasesError);
+    throwError(getCoApplicantCasesError.statusCode, getCoApplicantCasesError.message);
+  }
+
+  const concatAndDeDuplicateCases = (...cases) => [...new Set([].concat(...cases))];
+  return concatAndDeDuplicateCases(applicantCasesResult.Items, coApplicantCasesResult.Items);
+}
+
+async function getApplicantCases(personalNumber) {
+  const PK = `USER#${personalNumber}`;
+  const SK = 'CASE#';
+
+  const params = {
+    TableName: config.cases.tableName,
+    KeyConditionExpression: 'PK = :pk AND begins_with(SK, :sk)',
+    ExpressionAttributeValues: {
+      ':pk': PK,
+      ':sk': SK,
+    },
+  };
+
+  return dynamoDb.call('query', params);
+}
+
+async function getCoApplicantCases(personalNumber) {
+  const GSI1 = `USER#${personalNumber}`;
+  const SK = 'CASE#';
+
+  const params = {
+    TableName: config.cases.tableName,
+    IndexName: 'GSI1-SK-index',
+    KeyConditionExpression: 'GSI1 = :gsi1 AND begins_with(SK, :sk)',
+    ExpressionAttributeValues: {
+      ':gsi1': GSI1,
+      ':sk': SK,
+    },
+  };
+
+  return dynamoDb.call('query', params);
```

Comment:
I think it's fine sine we only define configuration stuff for params aka query parameters. But if anyone has a good idea of how this could be refactored without to much hassle, i'm all up for it (y)

Reply:
Perhaps something like (names subject to change) `getCase(personalNumber, caseId, gsiPersonalNumber, limit)`? Both `getCase` and `getCaseList` as well as a few other lambdas repeat similar code which could be extracted into a helper file.

Reply:
I think something along the lines of `getCase(userHash, caseHash, indexName)` would be more suitable

Reply:
Then we could do something like `getCase(buildUserHash(personaNumber), buildCaseHash(caseId), indexName)`

Reply:
Or even buildHashKey(prefix, value) => PREFIX#VALUE